src/main/java/com/redhat/devtools/lsp4ij/features/workspaceSymbol/AbstractLSPWorkspaceSymbolContributor.java

```diff
@@ -1,109 +1,130 @@
-        try {
-            waitUntilDone(workspaceSymbolFuture);
-        } catch (
-                ProcessCanceledException e) {//Since 2024.2 ProcessCanceledException extends CancellationException so we can't use multicatch to keep backward compatibility
-            //TODO delete block when minimum required version is 2024.2
-            workspaceSymbolSupport.cancel();
-            return null;
-        } catch (CancellationException e) {
-            // cancel the LSP requests workspace/symbol
-            workspaceSymbolSupport.cancel();
-            return null;
-        } catch (ExecutionException e) {
-            LOGGER.error("Error while consuming LSP 'workspace/symbol' request", e);
-            return null;
-        }
-
-        if (isDoneNormally(workspaceSymbolFuture)) {
-            // workspace/symbol has been collected correctly
-            return workspaceSymbolFuture.getNow(null);
-        }
-        return null;
-    }
-
+/*******************************************************************************
+ * Copyright (c) 2024 Red Hat, Inc.
+ * Distributed under license by Red Hat, Inc. All rights reserved.
+ * This program is made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution,
+ * and is available at https://www.eclipse.org/legal/epl-v20.html
+ *
+ * Contributors:
+ * Red Hat, Inc. - initial API and implementation
+ ******************************************************************************/
+package com.redhat.devtools.lsp4ij.features.workspaceSymbol;
+
+import com.intellij.ide.util.gotoByName.ChooseByNamePopup;
+import com.intellij.navigation.ChooseByNameContributorEx;
+import com.intellij.navigation.NavigationItem;
+import com.intellij.openapi.progress.ProcessCanceledException;
+import com.intellij.openapi.project.Project;
+import com.intellij.psi.search.GlobalSearchScope;
+import com.intellij.util.Processor;
+import com.intellij.util.indexing.FindSymbolParameters;
+import com.intellij.util.indexing.IdFilter;
+import com.redhat.devtools.lsp4ij.LSPWorkspaceSupport;
+import org.eclipse.lsp4j.WorkspaceSymbolParams;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.slf4j.Logger;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.CancellationException;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+
+import static com.redhat.devtools.lsp4ij.internal.CompletableFutures.isDoneNormally;
+import static com.redhat.devtools.lsp4ij.internal.CompletableFutures.waitUntilDone;
+
+/**
+ * Abstract base class for LSP goto symbol contributors
+ */
+abstract class AbstractLSPWorkspaceSymbolContributor implements ChooseByNameContributorEx {
+
+    private final Logger LOGGER;
+
+    protected AbstractLSPWorkspaceSymbolContributor(@NotNull Logger logger) {
+        this.LOGGER = logger;
```

Comment:
Don't set the LOGGER here. The logger should use the AbstractLSPWorkspaceSymbolContributor  LOGGER.

Reply:
I can certainly do that, but this is a pretty common idiom in inheritance hierarchies that allows each concrete subclass it have its own distinct logging channel that can be configured independently including inherited behavior. If you'd prefer that logging for both subclasses be configured via a single logger in `AbstractLSPWorkspaceSymbolContributor`, I can just remove the subclass loggers and restore the base class logger. Let me know your preference either way.

Reply:
yes please to be consistent with other code

Reply:
Thanks!